aws/resource_aws_launch_template_test.go

```diff
@@ -939,6 +940,108 @@ func TestAccAWSLaunchTemplate_hibernation(t *testing.T) {
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLaunchTemplateConfig_description(rName, description),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "default_version", "1"),
+					resource.TestCheckResourceAttr(resourceName, "latest_version", "1"),
+				),
+			},
+			// An updated config should cause a new version to be created
+			// but keep the default_version unchanged if unset
+			{
+				Config: testAccAWSLaunchTemplateConfig_description(rName, descriptionNew),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "default_version", "1"),
+					resource.TestCheckResourceAttr(resourceName, "latest_version", "2"),
+				),
+			},
+			// An updated config to set the default_version to an existing version
+			// should not cause a new version to be created
+			{
+				Config: testAccAWSLaunchTemplateConfig_descriptionDefaultVersion(rName, descriptionNew, 2),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "default_version", "2"),
+					resource.TestCheckResourceAttr(resourceName, "latest_version", "2"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSLaunchTemplate_updateDefaultVersion(t *testing.T) {
+	resourceName := "aws_launch_template.test"
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	description := "Test Description 1"
+	descriptionNew := "Test Description 2"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSLaunchTemplateDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLaunchTemplateConfig_description(rName, description),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "default_version", "1"),
+					resource.TestCheckResourceAttr(resourceName, "latest_version", "1"),
+				),
+			},
+			// Updating a field should create a new version but not update the default_version
+			// if update_default_version is set to false
+			{
+				Config: testAccAWSLaunchTemplateconfig_descriptionUpdateDefaultVersion(rName, descriptionNew, false),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "default_version", "1"),
+					resource.TestCheckResourceAttr(resourceName, "latest_version", "2"),
+				),
+			},
+			// Only updating the update_default_version field should not create a new version
+			// but use the default_version computed
+			{
+				Config: testAccAWSLaunchTemplateconfig_descriptionUpdateDefaultVersion(rName, descriptionNew, true),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "default_version", "1"),
+					resource.TestCheckResourceAttr(resourceName, "latest_version", "2"),
+				),
+			},
```

Comment:
I'm not sure if I'm misreading this test case, but as a practitioner, if I have a Launch Template that has `latest_version` deployed to a testing environment and `default_version` deployed to production, I think I would want configuring `update_default_version = true` and running `terraform apply` to promote the default version to match the latest version, without other changes or needing to lookup the latest and configure `default_version = 123` explicitly.

Reply:
ah that's a good catch! I cut out that logic 🤦‍♀️